File: src/viash/skeleton/template_test.R
library(testthat)

# create dummy input file
old_lines <- c("one", "two", "three")
writeLines(old_lines, "input.txt")

# run executable
system("./EXECUTABLE --input input.txt --output output.txt --option FOO")

# check whether output file exists
expect_true(file.exists("output.txt"))

# check whether content matches expected content
expected_lines <- c("FOO-one", "FOO-two", "FOO-three")
new_lines <- readLines("output.txt")
expect_equal(new_lines, expected_lines)

cat(">>> Test finished successfully!")

File: src/viash/skeleton/template_test.py
import unittest
import os
from os import path
import subprocess


with open("input.txt", "w") as writer:
    writer.writelines(["one\n", "two\n", "three\n"])


class MyTest(unittest.TestCase):
    def test_component(self):
        out = subprocess.check_output(["./EXECUTABLE", "--input", "input.txt", "--output", "output.txt", "--option", "FOO-"]).decode("utf-8")

        self.assertTrue(path.exists("output.txt"))
        
        with open("output.txt", "r") as reader:
            lines = reader.readlines()
        
        self.assertEqual(lines, ["FOO-one\n", "FOO-two\n", "FOO-three\n"])
    
        
unittest.main()
